src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -274,50 +315,92 @@ public void init(PlayerEntity player, ItemStack itemStack, Hand hand, CallbackIn
-        this.obfuscatedButton.visible = !this.signing;
+        Optional.ofNullable(boldButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(italicButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.visible = !this.signing);
 
         for (ColorSwatchWidget swatch : colorSwatches) {
-            swatch.visible = !this.signing;
+            if (swatch != null) {
+                swatch.visible = !signing;
+            }
         }
 
-        this.deletePageButton.visible = !this.signing && this.richPages.size() > 1;
-        this.insertPageButton.visible = !this.signing;
+        Optional.ofNullable(deletePageButton).ifPresent(button -> button.visible = !signing && richPages.size() > 1);
+        Optional.ofNullable(insertPageButton).ifPresent(button ->
+                button.visible = !signing && richPages.size() < MAX_PAGES_NUMBER
+        );
 
-        this.saveBookButton.visible = !this.signing;
-        this.loadBookButton.visible = !this.signing;
+        Optional.ofNullable(saveBookButton).ifPresent(button -> button.visible = !signing);
+        Optional.ofNullable(loadBookButton).ifPresent(button -> button.visible = !signing);
     }
 
     @Unique
-    private void updateState(@Nullable Formatting color, Set<Formatting> modifiers) {
-        boldButton.toggled = modifiers.contains(Formatting.BOLD);
-        italicButton.toggled = modifiers.contains(Formatting.ITALIC);
-        underlineButton.toggled = modifiers.contains(Formatting.UNDERLINE);
-        strikethroughButton.toggled = modifiers.contains(Formatting.STRIKETHROUGH);
-        obfuscatedButton.toggled = modifiers.contains(Formatting.OBFUSCATED);
-        setSwatchColor(color);
+    private void invalidateFormattingButtons() {
+        Optional.ofNullable(boldButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.BOLD));
+        Optional.ofNullable(italicButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.ITALIC));
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.UNDERLINE));
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.STRIKETHROUGH));
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.OBFUSCATED));
+        setSwatchColor(activeColor);
     }
 
     @Unique
     private void setSwatchColor(Formatting color) {
         for (ColorSwatchWidget swatch : colorSwatches) {
-            swatch.setToggled(swatch.getColor() == color);
+            if (swatch != null) {
```

Comment:
There should never be any nulls in the list, this is not needed.

Reply:
same reason as [above](https://github.com/chrrs/scribble/pull/18#discussion_r1760583931)

Reply:
Even if it's not initialized though, the list should be empty, and not contain nulls right?